### DISCUSSION TITLE
API Method Inputs with ":" in them

### DISCUSSION THREAD

**Discussion by achapkowski** [fastapi/fastapi#7011] [Questions] [answered]
### First check

* [x ] I added a very descriptive title to this issue.
* [ x] I used the GitHub search to find a similar issue and didn't find it.
* [ x] I searched the FastAPI documentation, with the integrated search.
* [ x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [ x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

I am trying to mimic an API's framework, and it requires some parameters to have a `:` in the name of the parameter:

Example:
```env:sr```

I can't name a parameter in the function with a **:** below, so I was wondering if there a way to specific the parameter name on the ```@app.post```, get... etc...

Thank you

Example code below:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.post("/", map_name={'env_sr' : 'env:sr'})
def read_root(env_sr:str):
    return {"Hello": env_sr}
```

**Comment by valentin994** [comment]
I'm not 100% sure what you are trying to do. So you want to pass an argument with the name env:sr or is it the value? 
The dead_root will return whatever param you send to it and it's expecting it to be a string so you can use ":" in the value.

**Comment by achapkowski** [comment]
I need to call my parameter **env:sr** , and I want the parameter name to show in the swagger doc as **env:sr**, so the user knows that this is the parameter name to pass.

**Comment by insomnes** [comment]
@achapkowski you can try `alias` parameter of `Query`

**Comment by insomnes** [comment, marked as answer]
@achapkowski yeah, tested it:
```
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/")
async def get_main(env_sr: str = Query(..., alias="env:sr")):
    return {"Hello": env_sr}
```

Works as expected: you can call the parameter like `/?env:sr=abc` and in docs it is shown as `env:sr`

**Comment by achapkowski** [comment]
it does work, thank you all!

**Comment by tiangolo** [comment]
Thanks for the help here @insomnes and @valentin994 ! 👏  🙇 

Thanks for reporting back and closing the issue @achapkowski  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
